script/model/model_server_test.py

```diff
@@ -0,0 +1,166 @@
+        """
+        Checks for necessary dependencies of the testsuite.
+        :return:
+        """
+        # 1. Check mini_runner
+        if not MINI_RUNNER_BINARY_PATH.exists():
+            raise FileNotFoundError(f"{str(MINI_RUNNER_BINARY_PATH)} not found. Please build mini_runner target!")
+
+        # 2. Check the bytecode_handlers_ir.bc exists
+        if not BYTECODE_HANDLER_IR_PATH.exists():
+            raise FileNotFoundError(f"{str(BYTECODE_HANDLER_IR_PATH)} not found. "
+                                    f"Please build hack_bytecode_handlers_ir target!")
+
+        # Copy that to the local directory (This hardcoded in llvm_engne.h:GetBytecodeHandlersBcPath)
+        copy2(BYTECODE_HANDLER_IR_PATH, os.getcwd())
+
+        # 3. Check test data is there
+        TEST_DATA_DIR.mkdir(exist_ok=True, parents=True)
+        TEST_SEQ_FILES_DIR.mkdir(exist_ok=True, parents=True)
+
+        # 4. Check the result dir exists
+        if not os.path.exists(MINI_TRAINER_RESULT_DIR):
+            os.mkdir(MINI_TRAINER_RESULT_DIR)
+
+
+    @classmethod
+    def tearDownClass(cls):
+        # Remove the bytecode_handlers_ir.bc
+        os.remove('bytecode_handlers_ir.bc')
+
+        # Remove all data generated
+        rmtree(MINI_TRAINER_RESULT_DIR)
+        rmtree(TEST_DATA_DIR)
+
+    def generate_seq_file(self):
+        """
+        Generate the trace files in TEST_SEQ_FILES_DIR
+
+        :return:  file path generated
+        """
+        benchmark_filters = '--benchmark_filter='+ 'SEQ0'
+        mini_runner_rows_limit = '--mini_runner_rows_limit=' + '10000'
+
+        args = [benchmark_filters, mini_runner_rows_limit]
+        try:
+            proc = subprocess.run(args=[str(MINI_RUNNER_BINARY_PATH)] + args, check=True, encoding='utf-8',
+                                  capture_output=True, timeout=TEST_TIMEOUT)
+        except subprocess.CalledProcessError as e:
+            self.fail(f'Running mini runner fails with errors: {e.stderr}')
+
+        # Move the seqfiles to test
+        return move('pipeline.csv', str(TEST_SEQ_FILES_DIR / 'SEQ0_execution.csv'))
+
+    def train_model(self, train_dir, result_path):
+        """
+        Train a mini_model
+        :param train_dir:  Where seq files are stored
+        :param result_path:  Result path for the mini_trainer
+        :return:
+        """
+        ml_models = ["lr", "rf", "gbm"]
+        test_ratio = 0.2
+        trim = 0.2
+        expose_all = True
+
+        trainer = MiniTrainer(train_dir, result_path, ml_models, test_ratio, trim, expose_all)
+        model_map = trainer.train()
+        return model_map
+
+
+    def test_inference(self):
+        """
+        Load the test model maps, and test if inference could be performed with the model
+        """
+
+        # Test parameters
+        file_path = self.generate_seq_file()
+        result_path = MINI_TRAINER_RESULT_DIR
+        model_map = self.train_model(str(TEST_SEQ_FILES_DIR), result_path)
```

Comment:
I would load the model from the dumped path instead of directly use the in-memory object. This will test the model saving and loading.